### PR TITLE
FEATURE : Add pagination in tasks page

### PR DESCRIPTION
**Pull request by sahsisunny**
## Depend on the below PR

- [ ]  https://github.com/Real-Dev-Squad/website-backend/pull/1232

**Test Coverage:**
image
image
image
image

### Description:
In this PR I have added pagination for fetching tasks. Now we can fetch tasks by passing the next and previous URLs in the request body.

### Anything you would like to inform the reviewer about:
none

### Dev Tested:
- [x] Yes

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/70854507/768a2776-14a6-4613-91f2-e9630b32a58c

### CODE REVIEW

File: src/app/services/tasksApi.ts [hunk 1]
```diff
@@ -1,37 +1,52 @@
...
         getAllTasks: builder.query<
-            TasksQueryResponse['tasks'],
-            { dev: boolean; status?: string }
+            TasksResponseType,
+            {
+                dev?: boolean;
+                status?: string;
+                nextPage?: string;
+                prevPage?: string;
+            }
```
Comment (bharati-21) [inline src/app/services/tasksApi.ts:20]: Please define a type separately

File: src/components/tasks/TasksContent.tsx [hunk 2]
```diff
@@ -24,21 +24,35 @@ export const TasksContent = () => {
...
 
     const {
-        data: tasks = [],
+        data: tasks = { tasks: [], next: '', prev: '' },
         isError,
         isLoading,
     } = useGetAllTasksQuery({
         dev: dev as boolean,
         status: activeTab,
+        nextPage: nextPage,
```
Comment (bharati-21) [inline src/components/tasks/TasksContent.tsx:37]: NIT: please use shorthand

File: src/pages/mine/index.tsx [hunk 3]
```diff
@@ -3,17 +3,17 @@ import Head from '@/components/head';
...
-import task from '@/interfaces/task.type';
+import { TasksResponseType } from '@/interfaces/task.type';
 import { LOGIN_URL } from '@/constants/url';
 import { NO_TASKS_FOUND_MESSAGE } from '@/constants/messages';
 import useAuthenticated from '@/hooks/useAuthenticated';
 import { useGetMineTasksQuery } from '@/app/services/tasksApi';
 import { Loader } from '@/components/tasks/card/Loader';
 
-function CardList({ tasks }: { tasks: task[] }) {
+function CardList({ tasks }: { tasks: TasksResponseType }) {
```
Comment (bharati-21) [inline src/pages/mine/index.tsx:13]: please destructure tasks from tasks props

File: src/pages/mine/index.tsx [hunk 4]
```diff
@@ -30,7 +30,7 @@ const Content = () => {
 
     if (isLoading) return <p>Loading...</p>;
     if (error) return <p>Something went wrong! Please contact admin</p>;
-    if (tasks && tasks.length)
+    if (tasks)
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/pages/mine/index.tsx:33]: Done

File: src/components/tasks/RenderTasksList.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+                ? {
+                      ...acc,
+                      [curr.status]: [...acc[curr.status as keyof task], curr],
+                  }
+                : { ...acc, [curr.status]: [curr] };
+        },
+        {}
+    );
+    console.log(tasksGroupedByStatus);
+
```
Comment (Pratiyushkumar) [inline src/components/tasks/RenderTasksList.tsx:25]: remove console.log please

File: src/components/tasks/TasksContent.tsx [hunk 6]
```diff
@@ -60,6 +84,24 @@ export const TasksContent = () => {
...
+                        Prev
+                    </button>
+                    <button
+                        className={classNames.paginationButton}
+                        onClick={fetchNextTasks}
+                        disabled={!tasksData.next}
+                    >
+                        Next
+                    </button>
+                </div>
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksContent.tsx:106]: Not needed for these small elements

File: src/components/tasks/TasksLoader/tasksloader.module.scss [hunk 7]
```diff
@@ -0,0 +1,55 @@
+.loader {
+    position: relative;
+    width: 100%;
+    height: 10rem;
+    margin-bottom: 10px;
+    border: 1px solid #d3d3d3;
+    padding: 15px;
+    background-color: #e3e3e3;
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksLoader/tasksloader.module.scss:8]: Done

File: src/app/services/tasksApi.ts (hunk 1 above)
Comment (sahsisunny) [reply src/app/services/tasksApi.ts:20]: Done

**Review by Pratiyushkumar — DISMISSED** [review]
LGTM

**Review by bharati-21 — CHANGES_REQUESTED** [review]

File: src/components/tasks/TasksContent.tsx [hunk 8]
```diff
@@ -40,26 +56,55 @@ export const TasksContent = () => {
...
+    if (isLoading) return <p>Loading...</p>;
+
+    if (isError) return <p>{TASKS_FETCH_ERROR_MESSAGE}</p>;
+
     return (
         <div className={classNames.tasksContainer}>
             <TabSection onSelect={onSelect} activeTab={activeTab} />
             <div>{renderTaskList()}</div>
+
+            {dev === 'true' ? (
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksContent.tsx:91]: I didn't get

[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksContent.tsx]: Done

File: src/components/tasks/TasksLoader/index.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+export const TasksLoader = () => {
+    return (
+        <div className={classNames.loader}>
+            <div className={classNames.wrapper}>
+                <div className={classNames.line}></div>
+                <div className={classNames.line}></div>
+                <div className={classNames.line}></div>
+                <div className={classNames.line}></div>
+                <div className={classNames.line}></div>
+                <div className={classNames.line}></div>
```
[the remark this answers is not in the document]
Comment (sahsisunny) [reply src/components/tasks/TasksLoader/index.tsx:12]: As per the discussion I am creating the referenced issue to skeleton look like task skeleton.

**Review by Pratiyushkumar — APPROVED** [review]
LGTM

**Review by vinit717 — APPROVED** [review]
LGTM

File: src/components/PaginationButton/index.tsx
Comment (bharati-21) [inline src/components/PaginationButton/index.tsx:1]: What I meant by create a separate component for button is a generic button component that takes text, change handler as props.

This is fine for now. But once this PR is done, we should create a generic button component and re-use it here.
That way you can create two buttons
```
<Button text="Prev" onClickHandler={handlePrevClicked} />
<Button text="Next" onClickHandler={handleNextClicked} />
```

File: src/components/PaginationButton/index.tsx [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+                onClick={fetchNext}
+                disabled={!hasNext}
+            >
+                {nextLabel}
+            </button>
+        </div>
+    );
+};
+
+export default PaginationButton;
```
Comment (bharati-21) [inline src/components/PaginationButton/index.tsx:47]: Kindly write test for this component.

**Review by bharati-21 — CHANGES_REQUESTED** [review]

**Review by bharati-21 — APPROVED** [review]

**Pull request merged**